### DISCUSSION TITLE
Why need to set Custom Headers?

### DISCUSSION THREAD

**Discussion by afzl-wtu** [ZoeyVid/NPMplus#340] [Q&A] [answered]
This Proxy does not forward real ip of the client to listening service. I need manually set custom location these headers:
```
   proxy_set_header X-Forwarded-Host $host:$server_port;
   proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
   proxy_set_header X-Forwarded-Proto $scheme;
```
But on orignal Nginx Proxy Manager we do not need to set headers instead it automatically forward the real ip in request to listening interface.

**Comment by Zoey2936** [comment, marked as answer]
https://github.com/ZoeyVid/nginx-proxy-manager/blob/develop/rootfs/usr/local/nginx/conf/conf.d/include/proxy.conf they are set

**Reply by afzl-wtu** [reply]
Is there any way or method we can see if Proxy is including those headers in http request to listener?
Are there any Docker container that can show real ip received in http headers to it?

**Reply by Zoey2936** [reply]
no idea at the moment

**Comment by Zoey2936** [comment]
did you maybe made other changes in the advanced tab?

**Reply by afzl-wtu** [reply]
No Advanced Tab is empty. Also I am using latest version. And MeshCentral container do not boot if I do not put those lines there.

**Comment by afzl-wtu** [comment]
MeshCentral requires these lines to get real ip of Request sender in order to drop tls and trust proxy. But If I remove those lines and save Then I get 502 bad gateway error. Also this error is presend when using Orignal jc21 NPM. I also add these lines there too for proper working of Meshcentral.

**Reply by Zoey2936** [reply]
so the bug exists on my fork and the original?

**Reply by afzl-wtu** [reply]
Yes I checked both fork and orignal today.

**Reply by Zoey2936** [reply]
can you please send me your compose file?

**Reply by afzl-wtu** [reply]
```
version: "3"
services:
    nginx-proxy-manager:
        container_name: nginx
        image: zoeyvid/nginx-proxy-manager
        restart: always
        network_mode: host
        volumes:
        - "./data:/data"
        - "./www:/var/www" # optional, if you
want to use it as webserver for html/php
#        - "/opt/npm-letsencrypt:/etc/letsencr
ypt" # Only needed for first time migration fr
om original nginx-proxy-manager to this fork
        environment:
        - "TZ=Asia/Karachi" # set timezone, de
fault UTC
#        - "PUID=1000" # set group id, default
 0 (root)
#        - "PGID=1000" # set user id, default
0 (root)
#        - "NIBEP=48693" # internal port, alwa
ys bound to 127.0.0.1, default 48693, you need
 to change it, if you want to run multiple npm
 instances in network mode host
#        - "NPM_PORT=81" # Port the NPM backen
d should be bound to, default 81, you need to
change it, if you want to run multiple npm ins
tances in network mode host
#        - "IPV4_BINDING=127.0.0.1" # IPv4 add
ress to bind, defaults to all
#        - "NPM_IPV4_BINDING=127.0.0.1" # IPv4
 address to bind for the NPM backend, defaults
 to all
#        - "IPV6_BINDING=[::1]" # IPv6 address
 to bind, defaults to all
#        - "NPM_IPV6_BINDING=[::1]" # IPv6 add
ress to bind for the NPM backend, defaults to
all
#        - "DISABLE_IPV6=true" # disable IPv6,
 incompatible with IPV6_BINDING, default false
#        - "NPM_DISABLE_IPV6=true" # disable I
Pv6 for the NPM backend, incompatible with NPM
_IPV6_BINDING, default false
#        - "NPM_LISTEN_LOCALHOST=true" # Bind
the NPM Dashboard on Port 81 only to localhost
, incompatible with NPM_IPV4_BINDING/NPM_IPV6_
BINDING/NPM_DISABLE_IPV6, default false
#        - "NPM_CERT_ID=1" # ID of cert, which
 should be used instead of dummycerts, default
 unset/dummycerts
#        - "DISABLE_HTTP=true" # disables ngin
x to listen on port 80, default false
#        - "NGINX_LOG_NOT_FOUND=true" # Allow
logging of 404 errors, default false
#        - "CLEAN=false" # Clean folders, defa
ult true
#        - "FULLCLEAN=true" # Clean unused con
fig folders, default false
#        - "PHP81=true" # Activate PHP81, defa
ult false
#        - "PHP81_APKS=php81-curl php-81-curl"
 # Add php extensions, see aviable packages he
re: https://pkgs.alpinelinux.org/packages?bran
ch=v3.17&repo=community&arch=x86_64&name=php81
-*, default none
#        - "PHP82=true" # Activate PHP82, defa
ult false
#        - "PHP82_APKS
```

**Reply by Zoey2936** [reply]
can you please run `docker exec -it nginx sh -c 'apk add --no-cache nano && nano /usr/local/nginx/conf/conf.d/include/proxy.conf'` and adjust the x-forwarded-host line to your needs and save then through ctrl+x, y, enter please and then restart the container please?

**Reply by Zoey2936** [reply]
and try again

**Reply by afzl-wtu** [reply]
Sure I will verify it later. But at the moment I have moved back to JC21 origanl NPM. It is working flawlessly and working without Custom section. It automatically forwards Header of real ip. echoip or meshcentral are working great.

**Reply by Zoey2936** [reply]
ok, but i thought it also does not work without the settings in the advanced tab on the original?

**Reply by afzl-wtu** [reply]
It was my mistake as I told you earlier that it does not work on orignal NPM too. But after Reinstalling I see it is working till now.

**Reply by Zoey2936** [reply]
but did you checked, if that helps?
